launcher_go/v2/docker/commands.go

```diff
@@ -0,0 +1,54 @@
+package docker
+
+import (
+	"context"
+	"github.com/discourse/discourse_docker/launcher_go/v2/config"
+	"github.com/discourse/discourse_docker/launcher_go/v2/utils"
+	"golang.org/x/sys/unix"
+	"io"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+type DockerBuilder struct {
+	Config   *config.Config
+	Ctx      *context.Context
+	Stdin    io.Reader
+	Dir      string
+	ImageTag string
+}
+
+func (r *DockerBuilder) Run() error {
+	if r.ImageTag == "" {
+		r.ImageTag = "latest"
+	}
+	cmd := exec.CommandContext(*r.Ctx, utils.DockerPath, "build")
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	cmd.Cancel = func() error {
+		return unix.Kill(-cmd.Process.Pid, unix.SIGINT)
+	}
+	cmd.Dir = r.Dir
+	cmd.Env = r.Config.EnvArray(false)
+	cmd.Env = append(cmd.Env, "BUILDKIT_PROGRESS=plain")
+	for k, _ := range r.Config.Env {
+		cmd.Args = append(cmd.Args, "--build-arg")
+		cmd.Args = append(cmd.Args, k)
+	}
+	cmd.Args = append(cmd.Args, "--no-cache")
+	cmd.Args = append(cmd.Args, "--pull")
+	cmd.Args = append(cmd.Args, "--force-rm")
```

Comment:
I see that this is a valid option but couldn't find the documentation for it. Is this used to forcefully stop the build?

Reply:
It was a legacy option, if running without buildkit:

```
DOCKER_BUILDKIT=0 docker build --help | grep force-rm
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.

      --force-rm                Always remove intermediate containers
```